src/charm.py

```diff
@@ -57,30 +56,26 @@ def __init__(self, *args):
         self._nrf_requires = NRFRequires(charm=self, relation_name=NRF_RELATION_NAME)
         self.unit.set_ports(PCF_SBI_PORT)
         self._certificates = TLSCertificatesRequiresV3(self, "certificates")
+        self.framework.observe(self.on.update_status, self._configure_sdcore_pcf)
         self.framework.observe(self.on.database_relation_joined, self._configure_sdcore_pcf)
         self.framework.observe(self.on.database_relation_broken, self._on_database_relation_broken)
         self.framework.observe(self._database.on.database_created, self._configure_sdcore_pcf)
         self.framework.observe(self.on.fiveg_nrf_relation_joined, self._configure_sdcore_pcf)
         self.framework.observe(self._nrf_requires.on.nrf_available, self._configure_sdcore_pcf)
         self.framework.observe(self._nrf_requires.on.nrf_broken, self._on_nrf_broken)
         self.framework.observe(self.on.pcf_pebble_ready, self._configure_sdcore_pcf)
+        self.framework.observe(self.on.certificates_relation_joined, self._configure_sdcore_pcf)
         self.framework.observe(
-            self.on.certificates_relation_created, self._on_certificates_relation_created
-        )
-        self.framework.observe(
-            self.on.certificates_relation_joined, self._on_certificates_relation_joined
+            self._certificates.on.certificate_available, self._configure_sdcore_pcf
         )
         self.framework.observe(
             self.on.certificates_relation_broken, self._on_certificates_relation_broken
         )
-        self.framework.observe(
-            self._certificates.on.certificate_available, self._on_certificate_available
-        )
         self.framework.observe(
             self._certificates.on.certificate_expiring, self._on_certificate_expiring
         )
 
-    def _configure_sdcore_pcf(self, event: EventBase):
+    def _configure_sdcore_pcf(self, event: EventBase) -> None:  # noqa C901
```

Comment:
Could we use a helper method similar to here not to get `too complex` warning ? https://github.com/canonical/sdcore-nrf-k8s-operator/blob/3e9a1777f7a9e035fe6e76768f30e134c1f8c226/src/charm.py#L153

Reply:
We got [this](https://github.com/canonical/sdcore-udm-k8s-operator/pull/58#discussion_r1497497542) comment on the same topic. I don't have a strong opinion, as it will be refactored anyway when we'll integrate `collect_status`.

Reply:
You are right, it will be refactored with collect_status.